rust/kernel/static_assert.rs

```diff
@@ -4,35 +4,198 @@
 
 /// Static assert (i.e. compile-time assert).
 ///
-/// Similar to C11 [`_Static_assert`] and C++11 [`static_assert`].
+/// There are several forms of this macro:
+///
+///   - Boolean assertion: `expr`.
+///   - Set membership assertion: `(expr) is in {a0, ..., aN}`.
+///   - Interval membership assertion: `(expr) is in [min, max]`.
+///   - Fits-in-type assertion: `(expr) fits in type`.
```

Comment:
When I saw this, the first thing I thought was whether a type could be 'transmuted' into another. For example, whether `*const T` fits in `usize`. Could we support something like this?

Reply:
So a comparison of the sizes? Or something more?

I wonder if it is best to leave that to utilities on their own (e.g. like `transmute` checks for the size).